### ISSUE TITLE
Search seems to cause errors in Firefox

### ISSUE THREAD

**Issue by gBasil** [krateng/maloja#237]
When I search for a track, Maloja sends a request to
`https://<domain>/api/search?max=5&query=a`
It returns a 308 to
`http://<domain>/apis/mlj_1/search?max=5&query=a`
And that returns a 301 to
`https://<domain>/apis/mlj_1/search?max=5&query=a`
And that request finally returns the search data. On Firefox, this ends up causing the request to not go through:
![a screenshot of the two requests in the Firefox Network panel in the devtools](https://github.com/krateng/maloja/assets/57513430/15965caf-0383-4c47-a772-6b275dca29e9)

I'm not sure if this is an error with my configuration (Maloja running on NixOS behind Cloudflare, accessed via Firefox) or if it's to do with Maloja redirecting to HTTP instead of HTTPS.

**Comment by Velocidensity** [comment]
Are you running maloja behind nginx? If so, you should set X-Forwarded-Proto to "https", otherwise maloja/bottle (the web server) has no way of knowing the user is using http. This bit of code in the proxy section does the trick:
```nginx
proxy_set_header X-Forwarded-Proto https;
```

Removing the redirects might be worthwhile (and has happened before, in #51), as there is no reason for the JS to have deprecated endpoints hardcoded (search, and rule toggling are two places I ran into this too).

**Comment by ThinkChaos** [comment]
I have the headers properly setup (confirmed by dumping them) but still have issues with plaintext redirects.
After looking into it, I'm pretty sure [bottlepy/bottle/pull/747](https://github.com/bottlepy/bottle/pull/747) would fix it but not fix the actual root cause.
The root cause is the HTTPS scheme info being lost, but I'm not sure how that happens.

I'm now using a workaround on the reverse proxy side.
In Nginx http block:
```
map $upstream_http_location $same_scheme_location {
  '~^http://(.+)' '//$1';
  default         $upstream_http_location;
}
```

In server or location block:
```
proxy_hide_header location;
add_header location $same_scheme_location;
```

**Comment by gBasil** [comment]
This issue started happening to me again as well, actually, but you seem to have actually diagnosed it (I just added a modified version of the search script as a userscript :P).

@ThinkChaos Should I reopen this issue?

**Comment by ThinkChaos** [comment]
Yeah I'd reopen it if it's also an issue for you. At least it'll make it easier for others to find this!
